lib/Directives/SubDirective.php

```diff
@@ -30,10 +31,44 @@ final public function process(
         string $data,
         array $options
     ): void {
-        $subParser = $parser->getSubParser();
+        if ($node === null) {
+            /*
+             * A directive with no content (which is not allowed for sub-directives.
+             *
+             * For example:
+             *
+             *      .. note::
+             *
+             *      Now more normal text.
+             */
+            throw new RuntimeException('Content expected, none found.');
```

Comment:
We actually had a test for this, but we were allowing it to create an empty note. Sphinx does not allow that: it issues a warning and renders no note.